plugins/woocommerce/src/Internal/Admin/Orders/Edit.php

```diff
@@ -0,0 +1,164 @@
+	}
+
+	/**
+	 * Setup hooks, actions and variables needed to render order edit page.
+	 *
+	 * @param \WC_Order $order Order object.
+	 */
+	public function setup( \WC_Order $order ) {
+		$this->order    = $order;
+		$current_screen = get_current_screen();
+		$current_screen->is_block_editor( false );
+		$this->screen_id = $current_screen->id;
+		$this->add_order_meta_boxes( $this->screen_id, __( 'Order', 'woocommerce' ) );
+		$this->add_order_specific_meta_box();
+
+		/**
+		 * From wp-admin/includes/meta-boxes.php.
+		 *
+		 * Fires after all built-in meta boxes have been added. Custom metaboxes may be enqueued here.
+		 *
+		 * @since 3.8.0.
+		 */
+		do_action( 'add_meta_boxes', wc_get_page_screen_id( 'shop-order' ), $this->order );
+		$this->enqueue_scripts();
+	}
+
+	/**
+	 * Hooks meta box for order specific meta.
+	 */
+	private function add_order_specific_meta_box() {
+		add_meta_box( 'postcustom', __( 'Custom Fields', 'woocommerce' ), array( $this, 'render_custom_meta_box' ), $this->screen_id, 'normal' );
+	}
+
+	/**
+	 * Render meta box for order specific meta.
+	 */
+	public function render_custom_meta_box() {
+		if ( ! isset( $this->custom_meta_box ) ) {
+			$this->custom_meta_box = new CustomMetaBox();
+		}
+		$this->custom_meta_box->output( $this->order );
+	}
+
+	/**
+	 * Render order edit page.
+	 */
+	public function display() {
+		$this->render_wrapper_start();
+		$this->render_meta_boxes();
+		$this->render_wrapper_end();
+	}
+
+	/**
+	 * Helper function to render wrapper start.
+	 *
+	 * @param string $notice Notice to display, if any.
+	 * @param string $message Message to display, if any.
+	 */
+	private function render_wrapper_start( $notice = '', $message = '' ) {
+		?>
+		<div class="wrap">
+		<h1 class="wp-heading-inline">
+			<?php echo esc_html( 'Edit order' ); ?>
+		</h1>
+		<?php
+		echo ' <a href="' . esc_url( admin_url( 'admin.php?page=wc-orders&action=new' ) ) . '" class="page-title-action"> Add order </a>';
+		?>
+		<hr class="wp-header-end">
+
+		<div id="poststuff">
+		<div id="post-body" class="metabox-holder columns-<?php echo ( 1 === get_current_screen()->get_columns() ) ? '1' : '2'; ?>">
+
+		<?php if ( $notice ) : ?>
+			<div id="notice" class="notice notice-warning"><p id="has-newer-autosave"><?php echo wp_kses_post( $notice ); ?></p></div>
+		<?php endif; ?>
+		<?php if ( $message ) : ?>
+			<div id="message" class="updated notice notice-success is-dismissible"><p><?php echo wp_kses_post( $message ); ?></p></div>
+			<?php
+		endif;
+	}
+
+	/**
+	 * Helper function to render meta boxes.
+	 */
+	private function render_meta_boxes() {
+		?>
+		<div id="postbox-container-1" class="postbox-container">
+			<?php do_meta_boxes( $this->screen_id, 'side', $this->order ); ?>
+		</div>
+		<div id="postbox-container-2" class="postbox-container">
+		<?php
+		do_meta_boxes( $this->screen_id, 'normal', $this->order );
+		do_meta_boxes( $this->screen_id, 'advanced', $this->order );
```

Comment:
An interesting thing: we are passing an order object to `do_meta_boxes()` here whereas, in the legacy editor, `do_meta_boxes()` receives `WP_Post` objects instead. 

Execution proceeds from `do_meta_boxes()` along to [`WC_Meta_Box_Order_Actions::output()`](https://github.com/woocommerce/woocommerce/blob/6.6.1/plugins/woocommerce/includes/admin/meta-boxes/class-wc-meta-box-order-actions.php#L25) and this method interacts with what it thinks is a post object, and tries to access [post object properties directly](https://github.com/woocommerce/woocommerce/blob/6.6.1/plugins/woocommerce/includes/admin/meta-boxes/class-wc-meta-box-order-actions.php#L31).

Everything still mostly works (though, I'm really only testing the rendering of course) but unfortunately it results in a lot of notices being generated since some of the properties that exist for `WP_Post` objects don't exist for order objects and, even where there is overlap, direct access to properties is a no-no for orders and results in doing-it-wrong notices.

In my testing, I see 39 notice-level errors (related to this issue) being emitted each time the editor is loaded. Apologies for not spotting this on my earlier pass.